thirdparty/text/CMakeLists.txt

```diff
@@ -0,0 +1,8 @@
+add_library(ztd.text ${PROJECT_SOURCE_DIR}/thirdparty/text/source/ztd/idk/bit.intrinsic.c
```

Comment:
Why not use [`CMAKE_CURRENT_SOURCR_DIR`](https://cmake.org/cmake/help/latest/variable/CMAKE_CURRENT_SOURCE_DIR.html)?